File: shared-atlas-dart/lib/src/interfaces/get_messages_response.dart
import 'package:shared_atlas_dart/shared_atlas_dart.dart';

/// Response structure for fetching messages.
/// 获取消息的响应结构。
class GetMessagesResponse {
  /// List of messages.
  /// 消息列表。
  final List<ChatMessageResponse> messages;
  /// Whether there are more older messages available.
  /// 是否还有更多旧消息。
  final bool hasMore;
  /// Cursor for the next page requests (use as 'before' param).
  /// 下一页请求的游标（作为 'before' 参数使用）。
  final String nextCursor;

  GetMessagesResponse({
    required this.messages,
    required this.hasMore,
    required this.nextCursor,
  });

  factory GetMessagesResponse.fromJson(Map<String, dynamic> json) {
    return GetMessagesResponse(
      messages: (json['messages'] as List<dynamic>).map((e) => ChatMessageResponse.fromJson(e as Map<String, dynamic>)).toList(),
      hasMore: json['hasMore'] as bool,
      nextCursor: json['nextCursor'] as String,
    );
  }
  Map<String, dynamic> toJson() {
    return {
      'messages': this.messages.map((e) => e.toJson()).toList(),
      'hasMore': this.hasMore,
      'nextCursor': this.nextCursor,
    };
  }
}

File: shared-atlas-rust/src/interfaces/get_messages_response.rs
use serde::{Serialize, Deserialize};
use crate::interfaces::chat_message_response::ChatMessageResponse;

/// Response structure for fetching messages.
/// 获取消息的响应结构。
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct GetMessagesResponse {
/// List of messages.
/// 消息列表。
    pub messages: Vec<ChatMessageResponse>,
/// Whether there are more older messages available.
/// 是否还有更多旧消息。
    pub has_more: bool,
/// Cursor for the next page requests (use as 'before' param).
/// 下一页请求的游标（作为 'before' 参数使用）。
    pub next_cursor: String,
}